docs/modules/ROOT/pages/explanations/decisions/forgejo.adoc

```diff
@@ -0,0 +1,346 @@
+<2> Used for the Helm Chart value `.gitea.admin.email`
+<3> This is passed to the Helm Chart at `.gitea`. Some values will be overwritten, some values will be added. To be figured out which exactly we allow.
+
+Injected configuration values:
+
+* `.gitea.admin`: Handled by the Composition and exposed in the connection secret.
+
+See <<_forgejo_by_vshn_2, default Helm values>>.
+
+=== Codey
+
+Codey only offers few configuration options, compared to the AppCat version.
+It also has pre-defined plans which set the compute and storage requests and limits accordingly.
+
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: Codey
+metadata:
+  name: my-codey-prod
+spec:
+  parameters:
+    service:
+      majorVersion: "9"
+      adminEmail: admin@example.com <1>
+    size: 
+      plan: mini <2>
+  writeConnectionSecretToRef:
+    name: codey-creds
+----
+<1> Used for the Helm Chart value `.gitea.admin.email`
+<2> Plans from https://www.codey.ch/#Beginner[Codey Website^]. See also <<Kubernetes Resource Configuration>>.
+
+Notes:
+
+* The FQDN is automatically set to `.metadata.name`.app.codey.ch
+* We use a default configuration for all Codey instances, for example we use Mailgun to send E-Mail
+
+See <<_codey_3, default Helm values>>.
+
+== Forgejo Configuration
+
+In this section the Forgejo configuration decisions are documented.
+
+=== Database
+
+There are several options for the database, most prominently PostgreSQL and SQLite.
+
+To begin with, we're using SQLite as the database backend, with the WAL mode.
+
+Once this is not enough anymore, we'll explore the options to use PostgreSQL as the backend and potentially migrate instances there should it be required.
+Migration to a different database can be done with the https://forgejo.org/docs/latest/admin/command-line/#dump[`forgejo dump`] command.
+
+Ref: https://forgejo.org/docs/latest/admin/recommendations/#databasedb_type[Recommended Settings and Tips: Database^]
+
+=== Storage
+
+We use local storage for everything, therefore a PVC is mounted to `/data`.
+
+While we could also use object storage for most of the data, it complicates a lot.
+For example backup would become an additional burden, as well as storage space monitoring also won't be that easy anymore.
+And it's not possible to store Git repositories on object storage, this way we'd have multiple storage locations.
+
+Ref: https://forgejo.org/docs/latest/admin/storage/[Admin documentation: Storage^]
+
+=== Cache
+
+There are several options for caching, most prominently Redis and TwoQueue.
+
+To begin with, we're using the `twoqueue` mechanisms with the recommended configuration for the `HOST` parameter:
+
+----
+{"size":100, "recent_ratio":0.25, "ghost_ratio":0.5} 
+----
+
+Once instances are growing, and we see issues with caching performance we'll switch to Redis by using https://products.vshn.ch/appcat/redis.html[Redis by VSHN^].
```

Comment:
I'd do the same here as for the databases. Small instance sizes will have the `TwoQueue` thingy and larger ones will have Redis. As this is only caching, migration shouldn't be an issue.

That way we shift the decision to the users.